components/Common/Search/States/WithSearchBox.tsx

```diff
@@ -78,15 +80,55 @@ export const WithSearchBox: FC<SearchBoxProps> = ({ onClose }) => {
     [searchTerm, selectedFacet]
   );
 
+  useKeyboardCommands(cmd => {
+    if (searchError || !searchResults || searchResults.count <= 0) return;
+
+    switch (true) {
+      case cmd === 'down' && selectedResult == null:
+        setSelectedResult(0);
+        break;
+      case cmd === 'down' &&
+        selectedResult != null &&
+        selectedResult < searchResults.count &&
+        selectedResult < DEFAULT_ORAMA_QUERY_PARAMS.limit - 1:
+        setSelectedResult(selectedResult + 1);
+        break;
+      case cmd === 'up' && selectedResult != null && selectedResult != 0:
+        setSelectedResult(selectedResult - 1);
+        break;
+      case cmd === 'enter':
+        handleEnter();
+        break;
+      default:
+    }
+  });
+
   const reset = () => {
     setSearchTerm('');
     setSearchResults(null);
+    setSelectedResult(null);
     setSelectedFacet(0);
   };
 
+  const handleEnter = () => {
+    if (selectedResult == null || (searchResults && searchResults?.count <= 0))
+      return;
+    const selectedHit = searchResults?.hits[selectedResult ?? 0];
+    if (!selectedHit) return;
+
+    router.push(searchHitToLinkPath(selectedHit));
+    reset();
```

Comment:
Shouldn't the searchbox be reset/closed before the route change?

Also, could you add the "reset" to be part directly of the onClose? Makes sense IMO for search results to always reset during onClose

Reply:
Yeah, it now uses a single `handleClose` function